test/TesterInternal/MembershipTests/MembershipTableTestsBase.cs

```diff
@@ -1,72 +1,76 @@
 using System;
+using System.Collections.Generic;
 using System.Linq;
 using System.Net;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Orleans;
-using Orleans.AzureUtils;
+using Orleans.Messaging;
 using Orleans.Runtime;
 using Orleans.Runtime.Configuration;
 using UnitTests.StorageTests;
 using Xunit;
-using Assert = Microsoft.VisualStudio.TestTools.UnitTesting.Assert;
 
 namespace UnitTests.MembershipTests
 {
-    [TestClass]
-    public abstract class MembershipTableTestsBase : IClassFixture<MembershipTableTestsBase.Fixture>, IDisposable
+    public abstract class MembershipTableTestsBase : IDisposable, IClassFixture<MembershipTableTestsBase.Fixture>
     {
-        public class Fixture : IDisposable
-        {
-            public Fixture()
-            {
-                TraceLogger.Initialize(new NodeConfiguration());
+        private static readonly string hostName = Dns.GetHostName();
+        private readonly TraceLogger logger;
+        private readonly IMembershipTable membershipTable;
+        private readonly IGatewayListProvider gatewayListProvider;
+        private readonly string deploymentId;
 
-                // Set shorter init timeout for these tests
-                OrleansSiloInstanceManager.initTimeout = TimeSpan.FromSeconds(20);
-            }
+        protected const string testDatabaseName = "OrleansMembershipTest";//for relational storage
 
-            public void Dispose()
-            {
-                // Reset init timeout after tests
-                OrleansSiloInstanceManager.initTimeout = AzureTableDefaultPolicies.TableCreationTimeout;
-            }
+        public class Fixture
+        {
+            public string ConnectionString;
         }
 
-        private static readonly string hostName = Dns.GetHostName();
-        private TraceLogger logger;
-        private IMembershipTable membershipTable;
-        private string deploymentId;
-
-        public MembershipTableTestsBase()
+        protected MembershipTableTestsBase(Fixture fixture)
         {
+            TraceLogger.Initialize(new NodeConfiguration());
             logger = TraceLogger.GetLogger(GetType().Name, TraceLogger.LoggerType.Application);
             deploymentId = "test-" + Guid.NewGuid();
 
             logger.Info("DeploymentId={0}", deploymentId);
 
+            lock (fixture)
```

Comment:
@shayhatsor You may know this, but locking widely available object bode ill luck. One explanation: http://stackoverflow.com/questions/11775205/why-is-it-a-bad-practice-to-lock-the-object-we-are-going-to-change. This looks fishy in that there are virtual calls in constructor and in addition the given parameter is used in a lock. Might not hurt now -- or ever -- but raises a spectre of suspicion upon reading. I would consider changing this, but on the other hand, I gather the test code is somewhat in flight currently and it might be beneficial to get this in. It might also be the Orleans maintainers are OK with this. @jdom, I think, is moving the test refactoring forwards. @sergeybykov may have an opinion.